dnsmasq: fix warning with poll.h include on musl

Warning is:
  #warning redirecting incorrect #include <sys/poll.h> to <poll.h>

Signed-off-by: Kevin Darbyshire-Bryant <kevin@darbyshire-bryant.me.uk>

--- a/src/dnsmasq.h
+++ b/src/dnsmasq.h
@@ -88,7 +88,7 @@ typedef unsigned long long u64;
 #if defined(HAVE_SOLARIS_NETWORK)
 #  include <sys/sockio.h>
 #endif
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/wait.h>
 #include <sys/time.h>
 #include <sys/un.h>
